File: src/main/java/team/zeromods/ancientmagic/item/MagicBook.java
package team.zeromods.ancientmagic.item;

import net.minecraft.client.Minecraft;
import net.minecraft.world.InteractionHand;
import net.minecraft.world.InteractionResultHolder;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;
import org.jetbrains.annotations.NotNull;
import team.zeromods.ancientmagic.client.MagicBookScreen;

public class MagicBook extends Item {
    public MagicBook() {
        super(new Properties().stacksTo(1));
    }

    @Override
    public @NotNull InteractionResultHolder<ItemStack> use(@NotNull Level level, @NotNull Player player,
                                                           @NotNull InteractionHand hand) {
        if (level.isClientSide) {
            Minecraft.getInstance().setScreen(new MagicBookScreen());
            return InteractionResultHolder.success(player.getItemInHand(hand));
        }

        return InteractionResultHolder.pass(player.getItemInHand(hand));
    }
}

File: src/main/java/team/zeromods/ancientmagic/item/MagicBook.kt
package team.zeromods.ancientmagic.item

import net.minecraft.client.Minecraft
import net.minecraft.world.InteractionHand
import net.minecraft.world.InteractionResultHolder
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.level.Level
import team.zeromods.ancientmagic.client.MagicBookScreen

class MagicBook : Item(Properties().stacksTo(1)) {
    override fun use(level: Level, player: Player, hand: InteractionHand): InteractionResultHolder<ItemStack> {
        return if (level.isClientSide) {
            Minecraft.getInstance().setScreen(MagicBookScreen())
            return InteractionResultHolder.success(player.getItemInHand(hand))
        } else InteractionResultHolder.pass(player.getItemInHand(hand))
    }
}